app/components/holiday-detail.js

```diff
@@ -37,12 +38,19 @@ export default Ember.Component.extend({
     hideEditForm() {
       this.get("day").rollbackAttributes();
       this.set("isEditing", false);
+      this.set("invalidName", false);
     },
 
     saveHoliday() {
       var holiday = this.get("day");
+      let isNamePresent = holiday.get("name").trim().length !== 0;
 
-      if (holiday.get("name").length !== 0) {
+      if (!isNamePresent) {
```

Comment:
you can add this as else part

      if (isNamePresent) {
        var loadingView = getOwner(this)
          .lookup("component:loading")
          .append();
        holiday
          .save()
          .catch(error => {
            holiday.rollbackAttributes();
            throw error;
          })
          .finally(() => {
            loadingView.destroy();
            this.set("isEditing", false);
            this.set("invalidName", false);
          });
      } esle{
      this.set("invalidName", true);
       return false;
       }